miner/worker.go

```diff
@@ -1075,29 +1112,64 @@ func (w *worker) fillTransactions(interrupt *int32, env *environment) error {
 			return err
 		}
 	}
+	if validatorCoinbase != nil && w.config.BuilderTxSigningKey != nil {
+		builderCoinbaseBalanceAfter := env.state.GetBalance(w.coinbase)
+		log.Info("Before creating validator profit", "validatorCoinbase", validatorCoinbase.String(), "builderCoinbase", w.coinbase.String(), "builderCoinbaseBalanceBefore", builderCoinbaseBalanceBefore.String(), "builderCoinbaseBalanceAfter", builderCoinbaseBalanceAfter.String())
+
+		profit := new(big.Int).Sub(builderCoinbaseBalanceAfter, builderCoinbaseBalanceBefore)
+		env.gasPool.AddGas(params.TxGas)
+		if profit.Sign() == 1 {
+			tx, err := w.createProposerPayoutTx(env, validatorCoinbase, profit)
+			if err != nil {
+				log.Error("Proposer payout create tx failed, validator payment skipped", "err", err)
```

Comment:
Shouldn't we error out of the function if the tx cannot be added, to avoid creating blocks without a proposer payment?
If it's causing problems with tests, you could make it configurable